modules/match.js

```diff
@@ -37,8 +47,20 @@ function match({ history, routes, location, ...options }, callback) {
     callback(
       error,
       redirectLocation,
-      nextState && { ...nextState, history, router }
+      nextState && {
+        ...nextState,
+        history,
+        router,
+        matchContext: { history, transitionManager, router }
```

Comment:
I get you're concerned that someone might have a mismatched `history`/`transitionManager`/`router` combo, but is that practically ever going to happen? This is entirely an internal-use, undocumented API, and if you use it, you'll need to learn how the code operates. In fact, you'll have to be reading through the code to even find it in the first place. 

So, this seems really weird to pass in the same things twice. It should just be the `router` and `transitionManager` props.